### ISSUE TITLE
Crop issues

### ISSUE THREAD

**Issue by kofa73** [darktable-org/darktable#17366]
### Describe the bug

Maybe these are two separate issues.
Issue 1: steps 2-3. Issue 2: steps 4-5.

According to `git bisect`, issue 2 is caused by:
```text
7a3c6f49f05b040a73a3763e053c3d674b56d9a1 is the first bad commit
commit 7a3c6f49f05b040a73a3763e053c3d674b56d9a1
Author: hanno@schwalm-bremen.de <hanno@schwalm-bremen.de>
Date:   Mon May 13 06:17:56 2024 +0200

    Fix late cropping issues
    
    "Original image" ratio and basically cropping for portrait images was broken.
    
    1. look for original image ratio in commit_params() and pass that correctly as d->aspect
       taking care of the aspect_rotation status
    2. In modify_roi_out() we must check for landscape vs portrait mode while keeping the aspect
    3. more informative log output via '-d pipe -d verbose'

 src/iop/crop.c | 59 +++++++++++++++++++++++++---------------------------------
 1 file changed, 25 insertions(+), 34 deletions(-)
```

### Steps to reproduce

1. Get https://tech.kovacs-telekes.org/files/2024-08-25-crop-issues/2024-08-22-17-57-27-P1040203.RW2 and open it in the darkroom.
2. Enable _crop_, select 3:1 panorama:
  ![image](https://github.com/user-attachments/assets/c50fde68-4dac-48a8-af9b-26a856747adb)
3. Move focus to another module.
  - Expectation: crop is applied.
  - Reality: crop module is disabled, crop is not applied.
4. Open _crop_. Manually drag the crop rectangle.
  ![image](https://github.com/user-attachments/assets/5bd5c69d-d8c9-471a-976b-19db3de8f353)
5. Move focus to another module. The crop is applied this time, but wrong:
  - Exceptation: image is cropped according to the crop rectangle.
  - Reality: the bottom right corner is not set according the crop rectangle; top left is OK:
  ![image](https://github.com/user-attachments/assets/330d549c-58af-4329-9fdd-2b4e516f6f53)

### Expected behavior

Apply selected crop correctly

### Commit

7a3c6f49f05b040a73a3763e053c3d674b56d9a1

### Where did you obtain darktable from?

self compiled

### darktable version

4.9.0+293~g9c5316789c

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 24.04

### Describe your system?

OpenCL on/off makes no difference.

### Are you using OpenCL GPU in darktable?

Yes

**Comment by jenshannoschwalm** [comment]
> 3\. Move focus to another module.
> 
> 
>     * Expectation: crop is applied.
> 
>     * Reality: crop module is disabled, crop is not applied.
> 
> 
>     4. Open _crop_. Manually drag the crop rectangle.

Hmm, i am not sure if your expectation is "good". This would mean whenever you expand the crop module (if it is not enabled right now) the current setting (original, 3:1 or whatever you are working in) would be applied and the module becomes enabled - whatever that is. I think it should stay disabled until either you do some change to the selection or actively enable the module with what-you-have. We could of course change that behaviour but it think that's good as it is.

**Comment by kofa73** [comment]
> select 3:1 panorama

So, I did change the state (settings) of the module, I didn't just expand it. It's quite confusing, I think, but if we document it, it's fine. The docs currently say:

> Resize the crop by dragging the border and corner handles.
> 
> Move the crop rectangle by clicking and dragging inside the crop area. Constrain movement to the horizontal/vertical axis by holding Ctrl/Shift, respectively while dragging. Commit changes by giving focus to another module.

So, they do mention changing the selected area, but clearly, not all of those operations are mandatory (it's OK to e.g. define the area, without subsequently resizing or moving it).

**Comment by kofa73** [comment]
4 & 5 are a separate issue, though, and a genuine bug, right?

**Comment by jenshannoschwalm** [comment]
It seems to me you are trying to be a "bad boy" by providing a quadratic raw-file :-) I couldn't think of such a cornercase yet - it seems the check for being either portrait or landscape fails here - so incorrect calculation to keep ratio. Will fix.

BTW i couldn't see this issue with other (non-quadratic) files, am i wrong here?

**Comment by kofa73** [comment]
> It seems to me you are trying to be a "bad boy" by providing a quadratic raw-file :-)

Nope, I was just so freaking tired that I did not realise the camera was in 1:1 mode (which I hardly ever use, so I probably moved the lever by mistake), even though I saw the image in the viewfinder. I noticed the image looked less wide than I expected (no wonder...), but by the time I figured out the cause, I had walked past the scene and was unable to summon the strength and will to go back.

**Comment by jenshannoschwalm** [comment]
Aah, i forgot my LX3 i had many ago also had that feature. (Unfortunately i did not shoot raws that time :-(
